### PR TITLE
add a method to test if an entity exist with some params

### PR DESCRIPTION
**Pull request by Charlie-Lucas**

### CODE REVIEW

**Review by alexisLefebvre — APPROVED** [review]

**Comment by lenybernard** [comment]
please consider implementing the inverse method `I should not find a User with lastname Abitbol`

File: src/Troopers/BehatContexts/Context/ExtendedEntityContext.php [hunk 1]
```diff
@@ -39,4 +40,57 @@ public function beforeScenario($event)
...
+
+        $row = array_shift($rows);
+        $object = $this->getEntityManager()
+            ->getRepository($entityName)
+            ->findOneBy(
+                $this->getQueryParams($entityName, $headers, $row)
+            );
+
+        if (is_null($object)) {
+            throw new \Exception(sprintf('There is not any object for the following params: %s', json_encode($this->getQueryParams($entityName, $headers, $row))));
```
Comment (lenybernard) [inline src/Troopers/BehatContexts/Context/ExtendedEntityContext.php]: What is it supposed to echo ? Something like this I guess :
```
There is not any object for the following params: {"firstname":"Abitlabol","enabled":false}
```

I guess we could be more pretty 💋  but it's ok for developers

**Review by lenybernard — CHANGES_REQUESTED** [review]
please consider implementing the inverse method:
```
Then I should not find one User like:
      | lastname |
      | Abitbol  |
```

**Review by lenybernard — CHANGES_REQUESTED** [review]
Please add a quantity variable instead of `one`,  We could be interested to find X users with lastname abitbol

File: src/Troopers/BehatContexts/Context/ExtendedEntityContext.php [hunk 2]
```diff
@@ -39,4 +40,88 @@ public function beforeScenario($event)
...
+     *
+     * @param $nbr
+     * @param $name
+     * @param TableNode $table
+     *
+     * @throws \Exception
+     */
+    public function existObjectLikeFollowing($nbr, $name, TableNode $table)
+    {
+        $entityName = $this->resolveEntity($name)->getName();
```
Comment (alexisLefebvre) [inline src/Troopers/BehatContexts/Context/ExtendedEntityContext.php]: The code from `$entityName` to `$this->getQueryParams($entityName, $headers, $row);` is duplicated, can you please factorize it in a method?

File: src/Troopers/BehatContexts/Context/ExtendedEntityContext.php [hunk 3]
```diff
@@ -39,4 +40,88 @@ public function beforeScenario($event)
...
+
+        $row = array_shift($rows);
+        $objects = $this->getEntityManager()
+            ->getRepository($entityName)
+            ->findBy(
+                $this->getQueryParams($entityName, $headers, $row)
+            );
+
+        if (count($objects) === 0) {
+            throw new \Exception(sprintf('There is not any %s for the following params: %s', $name, json_encode($this->getQueryParams($entityName, $headers, $row))));
```
Comment (alexisLefebvre) [inline src/Troopers/BehatContexts/Context/ExtendedEntityContext.php]: `$this->getQueryParams($entityName, $headers, $row)` is called 3 times in this method, a variable may be used to store this data.

**Review by lenybernard — APPROVED** [review]

**Pull request merged**